src/index.ts

```diff
@@ -96,7 +154,7 @@ export interface CreateSubscriptionCallback {
    apiResponse?: object): void;
 }
 
-export interface CreateSubscriptionOptions extends SubscriptionMetadata {
+export interface CreateSubscriptionOptions extends SubscriptionMetadataRaw {
```

Comment:
I don't think this is the right object to extend, name should never be required since all `createSubscription` methods have it as the first argument. And `topic.createSubscription` won't need the `topic` property either.

Reply:
Create Subscription need two field, messageRetentionDuration typeof number and pushEndpoint typeof string which is missing in SubscriptionMetadata, and included in SubscriptionMetadataRaw interface. let me know if any suggestion 👍

Reply:
Seems like our best course of action is probably not to extend any object but create a new one specifying the exact options the method accepts.

Reply:
done! hope you like this :)